_posts/2023-06-16-End-to-end-acknowledgements-in-Data-Prepper.md

```diff
@@ -0,0 +1,64 @@
+---
+layout: post
+title:  "End-to-end acknowledgements in Data Prepper"
+authors:
+- kkondaka
+- dlv
+date: 2023-06-16 10:00:00 -0700
+categories:
+  - technical-post
+
+meta_keywords: Data Prepper E2E, Data Prepper sink, Data Prepper pipeline as the sink
+meta_description: Learn how Data Prepper provides data durability through the use of end to end acknowledgements by delivering data to the sink before notifying the source.
+
+---
+
+## The need: Improving durability
+
+
+[Data Prepper](https://opensearch.org/docs/latest/data-prepper/index/) is an open-source data collector used to ingest data into OpenSearch clusters. To solve challenges with data durability, Data Prepper now has capability to use end-to-end acknowledgements.
+
+While Data Prepper has an in-memory buffer that allows for fast throughput, users also want improved data durability, particularly, confirmation that the data received by Data Prepper reaches the desired sink. Data Prepper maintainers and their teams have observed that a common challenge to data durability is the sink. For example, if the OpenSearch cluster cannot receive data because of temporary stress on the cluster or underscaling, then Data Prepper cannot send data to the destination sink. Further, Data Prepper may exhaust memory or other system resources or the hardware running Data Prepper may fail.
```

Comment:
The durability of the sink's cluster? Since we're referring to the OpenSearch cluster as the sink destination, correct?

Also, lets add this first sentence to the intro, per the preceding comment.

```suggestion
Data Prepper maintainers and their teams have observed that a common challenge to data durability is the durability sink's cluster. For example, if the OpenSearch cluster cannot receive data because of temporary stress on the cluster or underscaling, then Data Prepper cannot send data to the destination sink. Further, Data Prepper may exhaust memory or other system resources or the hardware running Data Prepper may fail.
```